src/drive/components/FileList.jsx

```diff
@@ -4,13 +4,46 @@ import { translate } from 'cozy-ui/react/I18n'
 import File, { FilePlaceholder } from '../containers/File'
 import LoadMore from './LoadMore'
 
+require('intersection-observer') // polyfill for safari
+
 const LIMIT = 30
 
 class FileList extends PureComponent {
   state = {
     isLoading: false
   }
 
+  intersectionObserver = null
+  loadMoreElement = null
+
+  componentWillMount() {
+    this.intersectionObserver = new IntersectionObserver(
+      this.checkIntersectionsEntries
+    )
+  }
+
+  checkIntersectionsEntries = intersectionEntries => {
+    if (intersectionEntries.filter(entry => entry.isIntersecting).length > 0) {
+      this.loadMoreRows(LIMIT)
+    }
+  }
+
+  updateLoadMoreElement = element => {
+    if (this.loadMoreElement) {
+      this.intersectionObserver.unobserve(this.loadMoreElement)
+    }
+
+    if (element && element.getDOMNode) {
+      this.loadMoreElement = element.getDOMNode()
```

Comment:
You shall use https://reactjs.org/docs/react-dom.html#finddomnode
as `getDOMNode` is deprecated since 0.14